src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentValidationServiceImpl.java

```diff
@@ -126,6 +138,78 @@ private boolean isExplanationValid(String explanation, CreateAccountingDocumentR
+                        buildValidationErrorMessageForUser(validationError));
+            }
+            LOG.info("documentWasParsedWithoutErrors: Detected one or more parsing validation errors.");
+            return false;
+        } else {
+            return true;
+        }
+    }
+    
+    private String buildValidationErrorMessageForUser(ValidationEvent validationError) {
+        String subMessage = getErrorSubMessage(validationError);
+        int lineNumber = getErrorLineNumber(validationError);
+        if (lineNumber != -1) {
+            return formatMessageFromProperty(CuFPKeyConstants.ERROR_CREATE_ACCOUNTING_DOCUMENT_XML_ADAPTER_ERROR,
+                    lineNumber, subMessage);
+        } else {
+            return subMessage;
+        }
+    }
+    
+    private String getErrorSubMessage(ValidationEvent validationError) {
+        String eventMessage = getErrorEventMessage(validationError);
+        LOG.warn("getErrorSubMessage: Detected validation error when parsing XML document: "
+                + eventMessage);
+        Matcher matcher = EXCEPTION_MESSAGE_PATTERN.matcher(eventMessage);
+        if (!matcher.matches()) {
+            return eventMessage;
+        }
+        
+        String exceptionClassname = matcher.group(EXCEPTION_CLASSNAME_GROUP);
+        if (StringUtils.isBlank(exceptionClassname)) {
+            throw new IllegalStateException(
+                    "Pattern matched message but it does not contain exception classname; this should NEVER happen!");
+        }
+        
+        String subMessage;
+        String detailMessage = matcher.group(DETAIL_MESSAGE_GROUP);
+        if (StringUtils.isNotBlank(detailMessage)) {
+            subMessage = detailMessage;
+        } else if (StringUtils.equals(NumberFormatException.class.getName(), exceptionClassname)) {
+            subMessage = getStringProperty(CuFPKeyConstants.ERROR_CREATE_ACCOUNTING_DOCUMENT_GENERIC_NUMERIC_ERROR);
+        } else {
+            subMessage = getStringProperty(CuFPKeyConstants.ERROR_CREATE_ACCOUNTING_DOCUMENT_GENERIC_ERROR);
+        }
+        
+        return subMessage;
+    }
+    
+    private String getErrorEventMessage(ValidationEvent event) {
+        String errorMessage = event.getMessage();
+        if (StringUtils.isBlank(errorMessage)) {
+            Throwable linkedException = event.getLinkedException();
+            if (linkedException != null) {
+                errorMessage = linkedException.getMessage();
+            }
+            if (StringUtils.isBlank(errorMessage)) {
+                errorMessage = getStringProperty(CuFPKeyConstants.ERROR_CREATE_ACCOUNTING_DOCUMENT_GENERIC_ERROR);
+            }
+        }
+        return errorMessage;
+    }
+    
+    private int getErrorLineNumber(ValidationEvent event) {
+        ValidationEventLocator locator = event.getLocator();
+        return locator != null ? locator.getLineNumber() : -1;
```

Comment:
is there a situation were ValidationEventLocator would not have the line number?  If so should the conditional also check that?

Reply:
According to the ValidationEventLocator Javadocs, the getLineNumber() method is supposed to return -1 if the line number is unavailable. Thus, I don't think any changes are needed here.

Reply:
Sounds good